src/coreclr/vm/method.hpp

```diff
@@ -1635,19 +1635,7 @@ class MethodDesc
     UINT16      m_wFlags3AndTokenRemainder;
 
     BYTE        m_chunkIndex;
-
-    enum {
-        // enum_flag2_HasPrecode implies that enum_flag2_HasStableEntryPoint is set.
-        enum_flag2_HasStableEntryPoint                  = 0x01,   // The method entrypoint is stable (either precode or actual code)
-        enum_flag2_HasPrecode                           = 0x02,   // Precode has been allocated for this method
-        enum_flag2_IsUnboxingStub                       = 0x04,
-        // unused                                       = 0x08,
-        enum_flag2_IsIntrinsic                          = 0x10,   // Jit may expand method as an intrinsic
-        enum_flag2_IsEligibleForTieredCompilation       = 0x20,
-        enum_flag2_RequiresCovariantReturnTypeChecking  = 0x40
-        // unused                                       = 0x80,
-    };
-    BYTE        m_bFlags2;
+    BYTE        m_methodIndex; // Used to hold the index into the chunk of this MethodDesc. Currently all 8 bits are used, but we could likely work with only 7 bits
```

Comment:
Does this comment really apply? It reads like this is a flags member, but the member is named `...Index` which is normally an integer table index.

Reply:
Hmmm. Perhaps my merge was invalid. Let me reevaluate this.

Reply:
Nope this is correct. The `m_methodIndex` was updated and the flags were removed. Note the -/+ before the source.

Reply:
I was referring to the ` Currently all 8 bits are used, but we could likely work with only 7 bits` portion, sorry to not have been more specific

Reply:
Ah. Normally I would agree with you because it is slightly out of place. The history of this data structure however makes the comment relevant. The `MethodDesc` data structure is designed to be as small and compact as possible, which means we place flags in odd places and consume bits where ever we can find them. The recent reorganization that added this field meant we were consuming all the bits in the current size of `MethodDesc`. The result is if we come up with a feature that "needs" a flag on the base `MethodDesc` there isn't going to be one in any obvious place. The intent of the comment is to inform future people that only 7 bits are needed and the high-order bit is likely available for use if needed.